src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Data/CollectionViewGroupInternal.cs

```diff
@@ -545,27 +545,20 @@ internal void AddSubgroupToMap(object nameKey, CollectionViewGroupInternal subgr
         private void RemoveSubgroupFromMap(CollectionViewGroupInternal subgroup)
         {
             Debug.Assert(subgroup != null);
+
             if (_nameToGroupMap == null)
-            {
                 return;
-            }
-            object keyToBeRemoved = null;
 
             // Search for the subgroup in the map.
-            foreach (object key in _nameToGroupMap.Keys)
+            foreach (KeyValuePair<object, WeakReference> item in _nameToGroupMap)
             {
-                WeakReference weakRef = _nameToGroupMap[key] as WeakReference;
-                if (weakRef != null &&
-                    weakRef.Target == subgroup)
+                if (item.Value.Target == subgroup)
```

Comment:
@h3xds1nz, why is the null check removed here ? I see that there is no way of updating the key, value pairs without calling `AddSubgroupToMap` and hence, the value is always a WeakReference, but that would be true earlier as well, but we still had the null check.

I kind of favor having the extra null checks rather than not having them.

Reply:
@dipeshmsft It's both because of the `as` casting pattern, now we use strongly-typed collection so there's no reason.

This was merely keeping PreSharp happy.